eva_submission/ena_retrieval.py

```diff
@@ -0,0 +1,120 @@
+import os.path
+
+import requests
+from ebi_eva_common_pyutils.config import cfg
+from ebi_eva_common_pyutils.logger import logging_config as log_cfg
+from ebi_eva_common_pyutils.metadata_utils import get_metadata_connection_handle
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query, execute_query
+
+from eva_submission.assembly_taxonomy_insertion import download_xml_from_ena
+
+logger = log_cfg.get_logger(__name__)
+
+
+def files_from_ena(search_term):
+    xml_root = download_xml_from_ena(f'https://www.ebi.ac.uk/ena/browser/api/xml/textsearch?result=analysis&query={search_term}')
+    analyses = xml_root.xpath('/ANALYSIS_SET/ANALYSIS')
+    analysis_files = {}
+    for analysis in analyses:
+        files = analysis.xpath('FILES/FILE')
+        file_dicts = []
+        for file in files:
+            file_dicts.append({'filename': file.attrib['filename'], 'filetype': file.attrib['filetype'],
+                               'md5': file.attrib['checksum'], 'analysis_accession': analysis.attrib['accession']})
+        analysis_files[analysis.attrib['accession']] = file_dicts
+    return analysis_files
+
+
+def get_file_id_from_md5(md5):
+    with get_metadata_connection_handle(cfg['maven']['environment'], cfg['maven']['settings_file']) as conn:
+        query = f"select file_id from file where file_md5='{md5}'"
+        rows = get_all_results_for_query(conn, query)
+        file_ids = [file_id for file_id, in rows]
+        if len(file_ids) > 1:
+            raise ValueError(f'Multiple file found with md5 {md5}')
+        elif len(file_ids) == 0:
+            return None
+        return file_ids[0]
+
+
+def insert_file_into_evapro(file_dict):
+    filename = os.path.basename(file_dict['filename'])
+    if file_dict['filename'].endswith('.vcf.gz') or file_dict['filename'].endswith('.vcf'):
+        file_type = 'vcf'
+    elif file_dict['filename'].endswith('.tbi'):
+        file_type = 'tabix'
+    else:
+        raise ValueError('Unsupported file type')
+    ftp_file = 'ftp.sra.ebi.ac.uk/vol1/' + file_dict['filename']
```

Comment:
I also see other entries for ftp_file attribute which start with "/ftp.ebi.ac.uk/pub/databases/eva/" for ex: "/ftp.ebi.ac.uk/pub/databases/eva/PRJEB45841/IPCT-FC218-MOON0057-Cap1690-25-ID22.vcf.gz". I am guessing these were incorrectly populated in the past?? If so, we should create a ticket to correct them...

Reply:
That's interesting. I checked and these FTP path don't seem to be used anywhere:
On the study browser for [PRJEB45841](https://www.ebi.ac.uk/eva/?eva-study=PRJEB45841). the links point correctly to the ENA browser for the submitted files and to our FTP for the browsable_files. But in both case I think the path is hard coded in eva-web.
I don't mind correcting them for consistency but right now they're not super useful.

Reply:
I see. Alright.